protocol/0019-MCAL-margin_calculator.md

```diff
@@ -42,21 +42,125 @@
+
+1. **Collateral search level**: when in cross-margin mode, the margin account balance below which the system will seek to recollateralise the margin account back to the initial margin level.
+
+1. **Initial margin**: when in cross-margin mode, the margin account balance initially allocated for the position, and the balance to which the margin account will be returned after collateral search and release, if possible.
+
+1. **Collateral release level**: when in cross-margin mode, the margin account balance above which the system will return collateral from a profitable position to the party's general account for potential use elsewhere.
+
+The protocol is designed such that ***Maintenance margin < Collateral search level < Initial margin < Collateral release level***.
+
+### Potential position margin level:
+
+1. **Order margin**: the amount of additional margin on top of the amount in the margin account that is required for the party's current active orders.
+Note that this may be zero if the active orders can only decrease the position size. 
 
 1. ***Maintenance margin***
 1. ***Collateral search level***
 1. ***Initial margin***
 1. ***Collateral release level***
 
-The protocol is designed such that ***Maintenance margin < Collateral search level < Initial margin < Collateral release level***.
 
 Margin levels are used by the protocol to ascertain whether a trader has sufficient collateral to maintain a margined trade. When the trader enters an open position, this required amount is equal to the *initial margin*. Subsequently, throughout the life of this open position, the minimum required amount is the *maintenance margin*. As a trader's collateral level dips below the *collateral search level* the protocol will automatically search for more collateral to be assigned to support this open position from the trader's general collateral accounts. In the event that a trader has collateral that is above the *collateral release level* the protocol will automatically release collateral to a trader's general collateral account for the relevant asset.
 
 **Whitepaper reference:** 6.1, section "Margin Calculation"
 
 In future there can be multiple margin calculator implementations that would be configurable in the market framework. This spec describes one implementation.
 
+
+
+## Isolated margin mode
+
+When in isolated margin mode, the position on the market has an associated margin factor.
+The margin factor must be greater than 0 and less than or equal to 1.
+
+Isolated margin mode can be enabled by placing an _update margin mode_ transaction.
+
+The default when placing an order with no change to margin mode specified must be to retain the current margin mode of the position.
+
+### Placing an order
+
+When submitting, amending, or deleting an order in isolated margin mode and continuous trading, a two step process will be followed:
```

Comment:
Maybe this is somehow implicitly implied somewhere, but: where exactly do we check the margin balances against the "maintenance x initial" level?